master/internal/api_trials.go

```diff
@@ -158,3 +158,52 @@ func (a *apiServer) KillTrial(
 	}
 	return &resp, err
 }
+
+func (a *apiServer) GetExperimentTrials(
+	_ context.Context, req *apiv1.GetExperimentTrialsRequest,
+) (*apiv1.GetExperimentTrialsResponse, error) {
+	resp := &apiv1.GetExperimentTrialsResponse{}
+
+	switch err := a.m.db.QueryProto("get_trials_for_experiment", &resp.Trials, req.ExperimentId); {
+	case err == db.ErrNotFound:
+		return nil, status.Errorf(codes.NotFound, "experiment %d not found:", req.ExperimentId)
+	case err != nil:
+		return nil, err
+	}
+	a.filter(&resp.Trials, func(i int) bool {
+		v := resp.Trials[i]
+		eliminate := false
+
+		if len(req.States) != 0 && !eliminate {
+			eliminate = true
+			for _, state := range req.States {
+				if state == v.State {
+					eliminate = false
+					break
+				}
+			}
+		}
+
+		return !eliminate
+	})
```

Comment:
The usage of `eliminate` is sort of confusing to read. Can we refactor this a bit; maybe we can eliminate `eliminate` and write it closer to the other filters (so if you've read one, skim them all easily).
```suggestion
	a.filter(&resp.Trials, func(i int) bool {
		v := resp.Trials[i]
		// If there are no filters, return true.
		if len(req.States) != 0 {
			return true
		}

		// Or if the filters are matched, return true.
		for _, state := range req.States {
			if state == v.State {
				return true
			}
		}

		// Else return false.
		return false
	})
```

Reply:
I thought I'm making it more readable :D

Reply:
in this case, we don't need a bool to keep track because there is really only one filter. but if there was more `eliminate` seem clearer to me (subjective) than the existing samples and my goal was to set a maybe more readable sample but it doesn't work as well since there is only one filter